test/util/velero/install.go

```diff
@@ -163,55 +167,89 @@ func VeleroInstall(ctx context.Context, veleroCfg *VeleroConfig, isStandbyCluste
 	return nil
 }
 
-// configvSpherePlugin refers to https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/v1.3.0/docs/vanilla.md
-func configvSpherePlugin(veleroCfg *VeleroConfig) error {
+// generateVSpherePlugin refers to
+// https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/v1.3.0/docs/vanilla.md
+func generateVSpherePlugin(veleroCfg *test.VeleroConfig) error {
 	cli := veleroCfg.ClientToInstallVelero
-	var err error
-	vsphereSecret := "velero-vsphere-config-secret"
-	configmaptName := "velero-vsphere-plugin-config"
-	if err := clearupvSpherePluginConfig(cli.ClientGo, veleroCfg.VeleroNamespace, vsphereSecret, configmaptName); err != nil {
-		return errors.WithMessagef(err, "Failed to clear up vsphere plugin config %s namespace", veleroCfg.VeleroNamespace)
-	}
-	if err := CreateNamespace(context.Background(), *cli, veleroCfg.VeleroNamespace); err != nil {
-		return errors.WithMessagef(err, "Failed to create Velero %s namespace", veleroCfg.VeleroNamespace)
+
+	if err := k8s.CreateNamespace(
+		context.Background(),
+		*cli,
+		veleroCfg.VeleroNamespace,
+	); err != nil {
+		return errors.WithMessagef(
+			err,
+			"Failed to create Velero %s namespace",
+			veleroCfg.VeleroNamespace,
+		)
 	}
+
+	clusterFlavor := "VANILLA"
```

Comment:
IMO instead of fallback based on error returned by "createxxxx" func, it's better to add another func to determine the flavor more explicitly.

Reply:
This can be improved in a subsequent PR.